### ISSUE TITLE
Trait or Stand-alone resource Controller

### ISSUE THREAD

**Issue by remoblaser** [remoblaser/laravel-resourceful#6]
Give the option to generate a Trait or a Stand-alone resource Controller instead of using the methods over and over again on every Controller

**Comment by landjea** [comment]
:+1: 

I might get a chance to do this one this weekend. Unless the sun comes out. :) (don't worry, it's Ontario, Canada so probably unlikely. :( ) I think the Trait should be the proper way to go, moving forward.

**Comment by remoblaser** [comment]
That would be awesome, thank you!

**Comment by landjea** [comment]
Tried on the weekend but missed something when thinking about this...we
have to include all of the models and the controllers on the trait. i.e:

<?php namespace {{namespace}}\Http\Controllers;

use {{namespace}}\Http\Requests;
use {{namespace}}\Http\Controllers\Controller;use Illuminate\Http\Request;
use {{namespace}}\Controller1;use
{{namespace}}\Http\Requests\Controller1Request;
use {{namespace}}\Controller2;use
{{namespace}}\Http\Requests\Controller2Request;
use {{namespace}}\Controller3;use
{{namespace}}\Http\Requests\Controller3Request;

trait ResourcefulTrait {

....

}

Unless I am missing something about Traits....? I thought maybe adding the
USE statements to the individual controllers would work but it didn't.
Perhaps using an Interface with the trait / controllers but then your
simple system is getting cumbersome....

**Comment by remoblaser** [comment]
Actually no, you don't need to use these.
You're defining a Generic Eloquent property / Illuminate Request etc. as a member, this needs to be set in the Controller constructor (to the Model / Request which should be used).

You then use those generic Objects to make calls etc., also in the functions you do not typecast your parameters.

I hope you know what i mean, would be able to give you a example until tomorrow.

**Comment by landjea** [comment]
Argh. I started with Php3 (yeah, a long time ago) and always forget about
the constructors to set those things.

I will give it a whack again when I get a chance tonight or tomorrow.

**Comment by landjea** [comment]
New wrinkle to consider : I was setting it up to create the trait if it
didn't already exist (we only need 1 trait). However if they create only a
subset of commands on first use then it will not have all 7 functions and
won't get created again. So now it will have to check which commands exist
when creating and if any are missing, tack them on the trait.

Just writing it down so I don't forget about it.

**Comment by remoblaser** [comment]
I think there should be a separate command, also we should include all the methods in the trait, if someone doesn't want to use Method X then he shouldn't use the trait and implement the methods on the Controller.

**Comment by landjea** [comment]
Agreed. Yeah, I don't know where my head was yesterday; definitely wasn't thinking right.

**Comment by landjea** [comment]
On that note, just got another project so I won't be looking at this for a few days...

**Comment by remoblaser** [comment]
Alright, i will take this over again then, thanks for investing time in it!

**Comment by landjea** [comment]
My apologies, I had thought I would be able to be more helpful. Good luck, great package.
